### PR TITLE
Add option to specify record to be read

### PR DESCRIPTION
**Pull request by martinbomio**
This PR adds an option to specify which record type should be read from the tfrecord files.
It also adds a new record type `PredictionLog`.

### CODE REVIEW

File: core/src/main/scala/tfr/instances/PredictionInstances.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  import io.circe._
+  import io.circe.parser._
+
+  private val ProtoPrinter =
+    JsonFormat.printer().omittingInsignificantWhitespace()
+
+  implicit val predictionLogEncoder: Encoder[PredictionLog] =
+    new Encoder[PredictionLog] {
+      override def apply(a: PredictionLog): Json =
+        parse(ProtoPrinter.print(a)).getOrElse(Json.fromString(""))
```
Comment (martinbomio) [inline core/src/main/scala/tfr/instances/PredictionInstances.scala:51]: for now, I used `JsonFormat` for parsing the json object since I didn't want to write an `Encoder` for each new type.

File: core/src/main/protobuf/tensorflow/core/example/example.proto [hunk 2]
```diff
@@ -0,0 +1,303 @@
+// Protocol messages for describing input data Examples for machine learning
```
Comment (andrewsmartin) [inline core/src/main/protobuf/tensorflow/core/example/example.proto:1]: I think this proto is actually published in `org.tensorflow % proto` so it doesn't need to be copied

File: core/src/main/protobuf/tensorflow/core/example/feature.proto [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// Protocol messages for describing features for machine learning model
```
Comment (andrewsmartin) [inline core/src/main/protobuf/tensorflow/core/example/feature.proto:1]: Same with this one

File: core/src/main/protobuf/tensorflow/core/example/example.proto (hunk 2 above)
Comment (martinbomio) [reply core/src/main/protobuf/tensorflow/core/example/example.proto:1]: I tried using the protobuf scope for those dependencies but for some reason, it was not finding them, let me try again

Comment (martinbomio) [reply core/src/main/protobuf/tensorflow/core/example/example.proto:1]: yeah tried adding:
```
tensorFlowProto % "protobuf",
```

also looked at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/maven/proto/pom.xml, and they don't seem to attach the `.proto` files to the jar

File: core/src/main/scala/tfr/instances/PredictionInstances.scala (hunk 1 above)
Comment (regadas) [reply core/src/main/scala/tfr/instances/PredictionInstances.scala:51]: Hmm, I see, yeah in the future it would be nice to have control over those, but not sure if it's worth it right now.

Would we it make sense to have `flat` feature over these?

Comment (martinbomio) [reply core/src/main/scala/tfr/instances/PredictionInstances.scala:51]: I can't think of any right now, there might be a need for flattening the inputs in the PredictionLong, but seems to me like we might be forcing it a bit
